### PR TITLE
MAINT: Remove remaining `numpy.int_` and filter `np.long` warning

### PR DESCRIPTION
**Pull request by mtsokol**
Hi @rgommers,

This is a follow-up PR for https://github.com/scipy/scipy/pull/19310.

I missed these two `int_`s because they were imported as `numpy.int_` instead of `np.int_`.

### CODE REVIEW

**Comment by rgommers** [comment]
Good catch. CI is unhappy though, looks like there's a bit more to clean up.

**Comment by mtsokol** [comment]
Ah yes, there was one more place - now it should be fine.

**Comment by mtsokol** [comment]
@rgommers @hawkinsp @pllim 

I pushed a change that additionally filters numpy warning in scipy.

File: scipy/_lib/_util.py [hunk 1]
```diff
@@ -37,8 +37,14 @@
 
 if np.lib.NumpyVersion(np.__version__) >= "2.0.0.dev0":
     try:
-        np_long = np.long  # type: ignore[attr-defined]
-        np_ulong = np.ulong  # type: ignore[attr-defined]
+        with warnings.catch_warnings():
+            warnings.filterwarnings(
+                action="ignore",
+                message=".*In the future `np.long` will be defined as.*",
```
Comment (pllim) [inline scipy/_lib/_util.py:43]: If you want to be more pedantic...
```suggestion
                message=r".*In the future `np\.long` will be defined as.*",
```

Thanks!

**Review by pllim — APPROVED** [review]
My approval does not matter but FWIW... Thanks for the quick patch!

**Comment by rgommers** [comment]
Sorry Mateusz, more trivia to work through - the CI failing on `test_warning_calls_filters` is real. The comment in that test explains - I think you can add this one to `allowed_filters`.

**Review by rgommers — APPROVED** [review]
Thanks for the updates @mtsokol, LGTM now. In it goes.

**Comment by rgommers** [comment]
@pllim thanks for the review! I kicked off new nightly builds, so the fix should propagate shortly.

**Comment by pllim** [comment]
I can confirm that the warning is now gone. Thanks again for the quick fix!

**Pull request merged**